src/Control/Carrier/NonDet/Church.hs

```diff
@@ -35,6 +35,10 @@ import Data.Functor.Identity
 
 -- | Run a 'NonDet' effect, using the provided functions to interpret choice, leaf results, and failure.
 --
+-- @
+-- runNonDet fork leaf nil ('pure' a '<|>' 'empty') = leaf a \`fork\` nil
+-- @
```

Comment:
I love how this single property completely encapsulates `NonDetC`’s behaviour w.r.t. its three continuations and the operators they interpret..